### DISCUSSION TITLE
does azuracast modify media files?

### DISCUSSION THREAD

**Discussion by ofsaleem** [AzuraCast/AzuraCast#6652] [Q&A] [answered]
i have a very large (15k or so songs) media library that's already tagged and organized. id love to point azuracast at the existing directory but i want to be sure it doesn't modify any of the files without explicit instruction. if there's a way to totally disable it from writing to them at all that would be even better

**Comment by ofsaleem** [comment]
i've found that i can't set the bind mount as `read_only: true` because azuracast is unable to create/write to the .albumart and .covers directories, and i can't configure them to be elsewhere. i think other than that issue id actually be able to set it to read only without issue

**Reply by BusterNeece** [reply, marked as answer]
@ofsaleem Often what we recommend users do is to go one directory deeper than the media directory and bind-mount _that_ instead, i.e. mount `/var/azuracast/stations/(your_station)/media/remote` as a read-only volume, that way we can still use the media storage location to store our cached metadata about the files, but your filesystem can remain read-only.

As for whether we actually write to media files, generally we do but only when a user edits the metadata via the web UI. We write those changes back to the file to ensure they persist, since most users expect that to be the case when editing tracks in our media editor. If you mount the filesystem as read-only, though, it'll just quietly fail that process but will still save metadata changes to our database.
